azext_edge/tests/edge/orchestration/test_instance_unit.py

```diff
@@ -104,3 +109,58 @@ def test_instance_list(mocked_cmd, resource_group_name: str, records: int):
     instances = Instances(mocked_cmd)
     result = list(instances.list(resource_group_name=resource_group_name))
     assert result == mock_instance_records["value"]
+    assert len(mocked_responses.calls) == 1
+
+
+@pytest.mark.parametrize(
+    "description",
+    [None, generate_random_string()],
+)
+@pytest.mark.parametrize(
+    "tags",
+    [None, {"a": "b", "c": "d"}],
+)
+def test_instance_update(mocked_cmd, mocked_responses: responses, description: Optional[str], tags: Optional[dict]):
+    instance_name = generate_random_string()
+    resource_group_name = generate_random_string()
+
+    mock_instance_record = get_mock_instance_record(name=instance_name, resource_group_name=resource_group_name)
+    mocked_responses.add(
+        method=responses.GET,
+        url=f"{BASE_URL}/subscriptions/{ZEROED_SUBSCRIPTION}/resourceGroups/{resource_group_name}/"
+        f"providers/{QUALIFIED_RESOURCE_TYPE}/{instance_name}?api-version={INSTANCES_API_VERSION}",
+        json=mock_instance_record,
+        status=200,
+        content_type="application/json",
+    )
+    mocked_responses.add(
+        method=responses.PUT,
+        url=f"{BASE_URL}/subscriptions/{ZEROED_SUBSCRIPTION}/resourceGroups/{resource_group_name}/"
+        f"providers/{QUALIFIED_RESOURCE_TYPE}/{instance_name}?api-version={INSTANCES_API_VERSION}",
+        json=mock_instance_record,
+        status=200,
+        content_type="application/json",
+    )
+
+    from azext_edge.edge.providers.orchestration.instances import Instances
```

Comment:
I see this got imported 3 times, wondering why don't we import once at the top of the file?

Reply:
Thinking behind this is the instance class is the subject of the test, so the test ensures a fresh instance

Reply:
Actually, going to change how these tests works in a next PR

Reply:
no sneak preview here :D?

Reply:
@Elsie4ever will move the import since it gets cached the first time